pkg/apis/rollouts/v1alpha1/analysis_types.go

```diff
@@ -193,6 +195,12 @@ type JobMetric struct {
 	Spec     batchv1.JobSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
 }
 
+// CloudWatchMetric defines the cloudwatch query to perform canary analysis
+type CloudWatchMetric struct {
+	Interval          DurationString `json:"interval,omitempty" protobuf:"bytes,1,opt,name=interval,casttype=DurationString"`
+	MetricDataQueries string         `json:"metricDataQueries" protobuf:"bytes,2,opt,name=metricDataQueries"`
```

Comment:
I think we can make the following improvement to the spec:

1. Instead of a string for `metricDataQueries`, I feel it would be better to make this field a list, where each element is a single `cloudwatch.MetricDataQuery`. This matches the AWS client API better.

In other words, I am proposing

```golang
type CloudWatchMetricDataQuery string

type CloudWatchMetric struct {
	Interval          DurationString `json:"interval,omitempty" protobuf:"bytes,1,opt,name=interval,casttype=DurationString"`
	MetricDataQueries []CloudWatchMetricDataQuery         `json:"metricDataQueries" protobuf:"bytes,2,opt,name=metricDataQueries"`
```

Then the spec would look like:

```yaml
    provider:
      cloudWatch:
        interval: 15m
        metricDataQueries:
        - |
            {
              "Id": "rate",
              "Expression": "errors / requests"
            }
        - |
            {
              "Id": "errors",
              "MetricStat": {
                "Metric": {
                  "Namespace": "app",
                  "MetricName": "errors"
                },
                "Period": 300,
                "Stat": "Sum",
                "Unit": "Count"
              },
              "ReturnData": false
            }
        - |
            {
              "Id": "requests",
              "MetricStat": {
                "Metric": {
                  "Namespace": "app",
                  "MetricName": "requests"
                },
                "Period": 300,
                "Stat": "Sum",
                "Unit": "Count"
              },
              "ReturnData": false
            }
```

2. I think we should allow `MetricDataQueries` to accept both JSON and YAML. So this would also be valid:

```yaml
    provider:
      cloudWatch:
        interval: 15m
        metricDataQueries:
        - |
            Id: errors
            MetricStat:
              Metric:
                Namespace: app
                MetricName: errors
              Period: 300
              Stat: Sum
              Unit: Count
            ReturnData: false
        - |
            Id: requests
            MetricStat:
              Metric:
                Namespace: app
                MetricName: requests
              Period: 300
              Stat: Sum
              Unit: Count
            ReturnData: false

```